proxy/proxy.go

```diff
@@ -495,3 +508,14 @@ func (proxy *Proxy) getDNSDomain() (domain string) {
 
 	return string(b)
 }
+
+func (proxy *Proxy) networkModeIsAttachable(networkMode string) (bool, error) {
+	for strings.HasPrefix(networkMode, "container:") {
+		container, err := proxy.client.InspectContainer(strings.TrimPrefix(networkMode, "container:"))
+		if err != nil || container.HostConfig == nil {
+			return false, err
+		}
+		networkMode = container.HostConfig.NetworkMode
+	}
+	return networkMode != "host", nil
+}
```

Comment:
There is a problem with this....

The parent container could be restarted with a different network mode. This is made worse by the fact that the reference can be a name, so you could actually change the parent completely.

My experiments indicate that this does not affect a running container. But does when the container is restarted. Which, if it happens w/o going through the proxy, e.g. due a restart policy, means we can end up with the wrong weavewait. Not much we can do about that, I suppose, but probably merits a comment.

Another edge case is that the parent container may not exist. Looks like we error at that point; shouldn't we leave that to docker? Though it is of course possible that the parent may appear by the time the request reaches docker. hmm.